Brokerages/Fxcm/FxcmBrokerage.Messaging.cs

```diff
@@ -387,9 +387,11 @@ private void OnExecutionReport(ExecutionReport message)
 
                         _isOrderSubmitRejected = true;
                     }
-
-                    _mapRequestsToAutoResetEvents[_currentRequest].Set();
-                    _mapRequestsToAutoResetEvents.Remove(_currentRequest);
+                    if (_mapRequestsToAutoResetEvents.ContainsKey(_currentRequest))
+                    {
```

Comment:
Maybe we could use `_mapRequestsToAutoResetEvents.TryGetValue` here, to have one less dictionary access ?